lambda/transl_array_comprehension.mli

```diff
@@ -0,0 +1,27 @@
+open Lambda
+open Typedtree
+open Debuginfo.Scoped_location
+
+(** Translate array comprehensions; see the .ml file for more details *)
+
+(** Translate an array comprehension ([Typedtree.comprehension], when it's the
+    body of a [Typedtree.Texp_array_comprehension]) into Lambda.  This generates
+    more efficient code in the case where the array has a known fixed size, by
+    preallocating the generated array; otherwise, it dynamically resizes the
+    generated array, cutting it back down to size at the end.  The [array_kind]
+    of the resulting array must be provided.
+
+    The only variables this term refers to are those that come from the array
```

Comment:
I don't know what you're referring to here.

Reply:
This comment describes the free variables of the term. Why? Why is this detail important enough to get a comment here?

Reply:
By way of contrast with `Transl_list_comprehension.comprehension`, which depends on various stdlib primitives.  It generally seems useful to me to know when you can use a lambda term (wrt free names), but I suppose none of these names are really truly "free" since they're in the stdlib.

Reply:
OK -- no worries then.